scripts/operations_tests/operations_tests.py

```diff
@@ -201,6 +206,49 @@ def load_monitor_log(self):
             log_lines = file.readlines()
             return InstrumentLog(log_lines, Path(log_file_path))
 
+    def create_fake_shutter_files(self):
+        shutter_count_data = np.array([[0, 74421], [1, 74420], [2, 74420], [3, 74420], [4, 74420], [5, 74418]])
+        shutter_times_data = np.array([[0, 0.01, 0.0001007], [1, 0.00032416, 0.00001], [2, 0.00031424, 0.00002],
+                                       [3, 0.00031424, 0.00003], [4, 0.00032256, 0.00002], [5, 0.00033472, 0.00004]])
+        spectra_file_data = np.column_stack((np.linspace(0.01, 0.0120275,
+                                                         num=100), np.linspace(1917858, 6389242, num=100, dtype=int)))
+
+        self.shutter_count_temp = tempfile.NamedTemporaryFile(suffix='_ShutterCount.txt', delete=False)
+        self.shutter_times_temp = tempfile.NamedTemporaryFile(suffix='_ShutterTimes.txt', delete=False)
+        self.spectra_file_temp = tempfile.NamedTemporaryFile(suffix='_Spectra.txt', delete=False)
```

Comment:
It might be better to create a `TemporaryDirectory`. Then you don't have the complication of `NamedTemporaryFile` being a file-like object.

```python
self.shutter_log_dir = tempfile.TemporaryDirectory()


np.savetxt(Path(self.shutter_log_dir.name) / "test_ShutterCount.txt", shutter_count_data, fmt='%.0f', delimiter='\t')
...
```

Then you might not need the `seek()`s and `delete_fake_shutter()` because the temporatry directory will be cleaned when the `self.shutter_log_dir` get deleted.